frontend/public/components/namespace.jsx

```diff
@@ -1044,106 +1110,104 @@ const namespaceBarDropdownDispatchToProps = (dispatch) => ({
-            removeQueryArgument('project-name');
-          },
-        });
-      } else {
-        onNamespaceChange && onNamespaceChange(newNamespace);
-        setActiveNamespace(newNamespace);
-        removeQueryArgument('project-name');
-      }
-    };
+  const items = data.map((item) => ({ title: item.metadata.name, key: item.metadata.name }));
 
-    return (
-      <div className="co-namespace-bar__items" data-test-id="namespace-bar-dropdown">
-        <Dropdown
-          disabled={disabled}
-          className="co-namespace-selector"
-          menuClassName="co-namespace-selector__menu"
-          buttonClassName="pf-m-plain"
-          canFavorite
-          items={items}
-          actionItems={defaultActionItem}
-          titlePrefix={model.label === 'Project' ? t('public~Project') : t('public~Namespace')}
-          title={title}
-          onChange={onChange}
-          selectedKey={activeNamespace || ALL_NAMESPACES_KEY}
-          autocompleteFilter={autocompleteFilter}
-          autocompletePlaceholder={
-            model.label === 'Project'
-              ? t('public~Select Project...')
-              : t('public~Select Namespace...')
-          }
-          userSettingsPrefix={NAMESPACE_USERSETTINGS_PREFIX}
-          storageKey={NAMESPACE_LOCAL_STORAGE_KEY}
-          shortCut={KEYBOARD_SHORTCUTS.focusNamespaceDropdown}
-        />
-        {children}
-      </div>
-    );
+  if (loaded && !items.find((option) => option.title === title) && title !== allNamespacesTitle) {
+    // Add current namespace if it isn't included
+    items.push({ title, key: title });
   }
-}
+
+  items.sort((a, b) => a.title.localeCompare(b.title));
+
+  if (canListNS) {
+    items.unshift({ title: allNamespacesTitle, key: ALL_NAMESPACES_KEY });
+  }
+
+  const onChange = (newNamespace) => {
+    if (newNamespace === CREATE_NEW_RESOURCE) {
+      createProjectModal({
+        blocking: true,
+        onSubmit: (newProject) => {
+          setActiveNamespace(newProject.metadata.name);
+          removeQueryArgument('project-name');
+        },
+      });
+    } else {
+      onNamespaceChange && onNamespaceChange(newNamespace);
+      setActiveNamespace(newNamespace);
+      removeQueryArgument('project-name');
+    }
+  };
+
+  return (
+    <div className="co-namespace-bar__items" data-test-id="namespace-bar-dropdown">
+      <NamespaceDropDown
+        options={items}
+        onSelect={(event, newNamespace) => {
+          onChange(newNamespace);
+        }}
+        selected={activeNamespace || ALL_NAMESPACES_KEY}
+        title={`${
+          model.label === 'Project' ? t('public~Project') : t('public~Namespace')
+        }: ${title}`}
+        actions={defaultActionItem}
+        userSettingsPrefix={NAMESPACE_USERSETTINGS_PREFIX}
+        storageKey={NAMESPACE_LOCAL_STORAGE_KEY}
+        isProjects={model.label === 'Project'}
+        disabled={disabled}
+        shortCut={KEYBOARD_SHORTCUTS.focusNamespaceDropdown}
+      />
```

Comment:
This can probably wait as a follow-on tech debt issue or something, but a lot of the state that we are handling in this component and passing down to NamespaceDropdown can actually just live in the NamespaceDropdown component itself. Previously, we had to manage the dropdown state here because we were using a more generic Dropdown component.